src/wallpaper.c

```diff
@@ -130,8 +130,10 @@ set_wallpaper (WallpaperDialogHandle *handle,
                const gchar *uri)
 {
   g_autoptr(GFile) source = NULL;
+  const gchar *path;
 
-  handle->picture_uri = g_build_filename (g_get_user_config_dir (), "background", NULL);
+  path = g_build_filename (g_get_user_config_dir (), "background", NULL);
+  handle->picture_uri = g_filename_to_uri (path, NULL, NULL);
```

Comment:
This value can be `NULL` now. The `g_file_new_for_uri` above does not handle this and it silently fails here.

Reply:
Do you think we could simply `g_assert (uri != NULL)`  above g_file_new_for_uri as it would be considered a client error to provide an invalid uri?

Reply:
Well it wouldn't be a client error. It would be a user using an encoding with characters that cannot be represented in UTF-8.

This should be extremely rare but possible. I would handle the `GError`, print out a warning, and return without doing anything.

Reply:
Asserting can never be right - it'll make the portal crash on a client error